app/controllers/samples_controller.rb

```diff
@@ -181,6 +183,31 @@ def bulk_upload
     end
   end
 
+  # POST /samples/bulk_upload_with_metadata
+  # Used to bulk-upload samples with remote input files in S3 directory.
+  # TODO(mark): Handle required metadata fields.
+  def bulk_upload_with_metadata
+    samples_to_upload = samples_params || []
+    metadata = params[:metadata] || {}
+    editable_project_ids = current_power.updatable_projects.pluck(:id)
+
+    # Ignore samples that aren't part of editable projects.
+    # TODO(mark): Show error for samples that aren't part of project.
+    samples_to_upload = samples_to_upload.select { |sample| editable_project_ids.include?(Integer(sample["project_id"])) }
+
+    errors, samples = upload_samples_with_metadata(samples_to_upload, metadata).values_at("errors", "samples")
+
+    respond_to do |format|
+      if errors.empty? && !samples.empty?
+        # tags = %W[client:web type:bulk user_id:#{current_user.id}]
+        # MetricUtil.put_metric_now("samples.created", @samples.count, tags)
```

Comment:
ah you can leave these in for now until we get the new dashboards all set up

Reply:
Oops! I didn't mean to remove this.